zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/store/OpensearchBatchRequest.java

```diff
@@ -0,0 +1,248 @@
+      final String index,
+      final String id,
+      final ExporterEntity entity,
+      final Map<String, Object> updateFields,
+      final String routing) {
+    LOGGER.debug(
+        "Add upsert request with routing {} for index {} id {} entity {} and update fields {}",
+        routing,
+        index,
+        id,
+        entity,
+        updateFields);
+
+    bulkRequestBuilder.operations(
+        op ->
+            op.update(
+                upd ->
+                    upd.index(index)
+                        .id(id)
+                        .upsert(entity)
+                        .document(updateFields)
+                        .routing(routing)));
+
+    return this;
+  }
+
+  @Override
+  public BatchRequest upsertWithScript(
+      final String index,
+      final String id,
+      final ExporterEntity entity,
+      final String script,
+      final Map<String, Object> parameters) {
+    return upsertWithScriptAndRouting(index, id, entity, script, parameters, null);
+  }
+
+  @Override
+  public BatchRequest upsertWithScriptAndRouting(
+      final String index,
+      final String id,
+      final ExporterEntity entity,
+      final String script,
+      final Map<String, Object> parameters,
+      final String routing) {
+    LOGGER.debug(
+        "Add upsert request with routing {} for index {} id {} entity {} and script {} with parameters {} ",
+        routing,
+        index,
+        id,
+        entity,
+        script,
+        parameters);
+
+    bulkRequestBuilder.operations(
+        op ->
+            op.update(
+                upd ->
+                    upd.index(index)
+                        .id(id)
+                        .upsert(entity)
+                        .script(scriptBuilder.getScriptWithParameters(script, parameters))
+                        .routing(routing)
+                        .retryOnConflict(UPDATE_RETRY_COUNT)));
+
+    return this;
+  }
+
+  @Override
+  public BatchRequest update(
+      final String index, final String id, final Map<String, Object> updateFields) {
+    LOGGER.debug(
+        "Add update request for index {} id {} and update fields {}", index, id, updateFields);
+
+    bulkRequestBuilder.operations(
+        op ->
+            op.update(
+                upd ->
+                    upd.index(index)
+                        .id(id)
+                        .document(updateFields)
+                        .retryOnConflict(UPDATE_RETRY_COUNT)));
+
+    return this;
+  }
+
+  @Override
+  public BatchRequest update(final String index, final String id, final ExporterEntity entity) {
+    LOGGER.debug("Add update request for index {} id {} and entity {}", index, id, entity);
+
+    bulkRequestBuilder.operations(
+        op ->
+            op.update(
+                upd ->
+                    upd.index(index).id(id).document(entity).retryOnConflict(UPDATE_RETRY_COUNT)));
+
+    return this;
+  }
+
+  @Override
+  public BatchRequest updateWithScript(
```

Comment:
I feel like you could also condense the update to one method call similar to the above approaches.